frontend/src/components/app.jsx

```diff
@@ -0,0 +1,49 @@
+// for refactoring, compare with authentication example here:
+// https://github.com/auth0/auth0-react/blob/gh-pages/examples/redirect-lock-with-api/app.jsx
+
+import React from 'react/addons';
+import {RouteHandler, Navigation} from 'react-router';
+import lockGenerator from '../lock-generator';
+import Home from './home/home.jsx';
+
+let App = React.createClass({
+  mixins: [Navigation],
+
+  componentWillMount() {
+    this.createLock();
+    this.setState(
+      {
+        idToken: localStorage.getItem('userToken'),
+        userProfile: localStorage.getItem('userProfile')
+      }
+    );
+  },
+
+  createLock() {
+    this.lock = lockGenerator();
+  },
+
+  onLogin() {
+    this.setState({
+      idToken: localStorage.getItem('userToken')
+    });
+    this.transitionTo('dashboard');
+  },
+
+  render() {
+    if (this.state.idToken) {
+      return (
+        <div className='appContainer'>
```

Comment:
This again raises the (unimportant) question of whether to render the React app to a div with id="app" like we have here:

https://github.com/the-ninja-turtles/turtle/blob/develop/frontend/server/views/index.ejs#L10

or whether to render it directly into the body. If we render it directly into the body, then the classname of out App component should be `'app'`